pkg/userdata/ubuntu/userdata.go

```diff
@@ -178,37 +178,189 @@ write_files:
   content: |
 {{ if ne .CloudConfig "" }}{{ .CloudConfig | indent 4 }}{{ end }}
 
+- path: "/etc/apt/sources.list.d/docker.list"
+  permissions: "0644"
+  content: deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable
+
+- path: "/etc/apt/sources.list.d/projectatomic-ubuntu-ppa-xenial.list"
+  permissions: "0644"
+  content: |
+    deb http://ppa.launchpad.net/projectatomic/ppa/ubuntu xenial main
+
+- path: "/usr/local/bin/install-dependencies"
+  permissions: "0777"
+  content: |
+    #!/bin/bash
+    set -xeuo pipefail
+    apt-key add /opt/docker.asc
+    apt-key add /opt/crio.asc
+    apt update
+
+    {{- if .OSConfig.DistUpgradeOnBoot }}
+    apt dist-upgrade -y
+    {{- end }}
+    if [[ -e /var/run/reboot-required ]]; then
```

Comment:
Actually I think this is still a problem, because:

* The `install-depencies` will ultimatively be kicked of by the `kubeadm-join.service`
* `kubeadm-join.service` gets kicked off by cloud init which will only happen on the first boot, not after the reboot


We can however not just activate `kubeadm-join.service` because it will fail on subsequent attempts.

Not sure what the best approach here is. Make `kubeadm-join.service` idempotent by trying to "somehow" find out if the kubelet is already joined and not execute the join in that case and just quit?